OpenRA.Game/SelectableExts.cs

```diff
@@ -28,7 +28,18 @@ public static int SelectionPriority(this ActorInfo a)
 
 		public static int SelectionPriority(this Actor a)
 		{
+			var modifiers = Game.GetModifierKeys();
+			var selectAllMode = modifiers.HasModifier(Modifiers.Ctrl);
+			var selectLowPriorityMode = modifiers.HasModifier(Modifiers.Alt);
+
 			var basePriority = a.Info.TraitInfo<SelectableInfo>().Priority;
+
+			if (selectAllMode && a.Info.TraitInfo<SelectableInfo>().IncludeInSelectAllMode)
+				basePriority = 10;
+
+			if (selectLowPriorityMode && !a.Info.TraitInfo<SelectableInfo>().IncludeInSelectLowPriorityMode)
```

Comment:
at this point it probably makes sense to store `SelectableInfo` in a variable

Reply:
Thanks for the suggestion. Implemented.